tembo-pgmq-python/Makefile

```diff
@@ -1,19 +1,30 @@
 SCOPE=tembo_pgmq_python/
-POETRY_VERSION:=1.7.1
+MIN_POETRY_VERSION=1.7.1
 
-format:
+.PHONY: check-poetry-version format lints clear-postgres run-pgmq-postgres test
+
+check-poetry-version:
+	@poetry_version=$$(poetry --version | awk '{print $$3}'); \
+	required_version=$(MIN_POETRY_VERSION); \
+	if [ $$(printf '%s\n' "$$required_version" "$$poetry_version" | sort -V | head -n1) != "$$required_version" ]; then \
+		echo "Poetry version $$poetry_version is not sufficient. Please install Poetry version >= $(MIN_POETRY_VERSION)."; \
+		exit 1; \
+	fi
+
+format: check-poetry-version
```

Comment:
Thoughts on managing the project with https://asdf-vm.com/? Not in the PR, but in the future. I find it very useful for managing the various toolchain versions, like python and poetry.

Reply:
did you use it with poetry directly? 
I update my poetry and add version control for sake of lower version, but if it has good impact on developing experience I will be glad to use it.

Reply:
I install python and poetry using asdf, like this.

```
asdf install poetry 1.7.1
asdf install python 3.11.0
```

Then, specify the versions in the `.tool-versions` file. 

```text
python 3.11.0
poetry 1.7.1
```

`.tool-versions` can be committed to source control, but it is currently not for this project. I just found it useful for managing python virtual environments.